app/src/main/java/com/sdp13epfl2021/projmag/database/UserProfileDatabase.kt

```diff
@@ -59,69 +65,63 @@ class UserProfileDatabase(
         }
     }
 
-    public fun getProfile(onSuccess: (profile: ImmutableProfile?) -> Unit) {
+    /**
+     * Gets the user profile from firebase if it exists
+     * @param onSuccess the closure that's when a profile is downloaded successfully with the fetched profile passed to it
+     * @param onFailure the closure that's called if the download fails
+     */
+    public fun getProfile(
+        onSuccess: (profile: ImmutableProfile?) -> Unit,
+        onFailure: (Exception) -> kotlin.Unit
+    ) {
         val userUid = getUser()?.uid
         if (userUid != null) {
             val docRef = firestore.collection(ROOT).document(userUid)
             docRef.get()
                 .addOnSuccessListener { document ->
                     if (document != null) {
-                        val firstName = document["firstName"] as? String
-                        val lastName = document["lastName"] as? String
-                        val ageLong = document["age"]
+                        val firstName = (document["firstName"] as? String)
+                        val lastName = (document["firstName"] as? String)
+                        val age = (document["age"] as? Long)?.toInt()
+                        val sciper = (document["sciper"] as? Long)?.toInt()
+                        val phoneNumber = (document["phoneNumber"] as? String)
 
-                        val gender = when (document["gender"] as? String) {
+                        val gender = when ((document["gender"] as? String)) {
                             Gender.MALE.name -> Gender.MALE
                             Gender.FEMALE.name -> Gender.FEMALE
                             else -> Gender.OTHER
                         }
-                        val phoneNumber = document["phoneNumber"] as? String
-                        val roleString = document["role"] as? String
-                        val sciperLong = document["sciper"]
 
-                        val role = when (roleString) {
+                        val role = when (document["role"] as? String) {
                             Role.TEACHER.name -> Role.TEACHER
                             Role.STUDENT.name -> Role.STUDENT
                             else -> Role.OTHER
```

Comment:
If the field of role is null or invalid, this line will return it as Role.OTHER.
Should an invalid role be equals to OTHER ?

Reply:
Yes